### ISSUE TITLE
Training exits after warnings

### ISSUE THREAD

**Issue by mariam-162** [yiyulics/CSEC#12]
Thanks for the complete code to help understand your paper.
I can run the test script using the given pre-trained model. But cannot train using: python src/train.py name=your_experiment_name

I'm using LCDP dataset. The code exits after the following warnings:
![Capture1](https://github.com/user-attachments/assets/f7462573-6ea9-495a-b503-9e107030bd72)
![Capture2](https://github.com/user-attachments/assets/c010282e-ff1e-47f7-8466-3c8d29761110)

LCDP dataset and pre-trained model were downloaded from here:
https://drive.google.com/drive/folders/10Reaq-N0DiZiFpSrZ8j5g3g0EJes4JiS

Command:
python src/train.py name=your_experiment_name

### Platform
- OS & OS Version: Windows 10 
- GPU: RTX 3070
- GPU Driver Version: 560.70
- RAM (GB): 64GB

I'm attaching complete testing and training logs for your reference. 

Issue # 2:
Apparently, GPU isn't being used for 'training'. But it is being used for testing.
[test_log.txt](https://github.com/user-attachments/files/18326771/test_log.txt)
 Can you please check what I might be doing wrong here? Maybe configurations need to be updated. Please help!
[train_log.txt](https://github.com/user-attachments/files/18326772/train_log.txt)

**Comment by yiyulics** [comment]
> The code exits after the following warnings

The warnings are not significant. The *libpng* warnings mainly relate to strict regulations regarding ICC profile checks in the latest `libpng` package. It is safe to ignore them. If you prefer not to see these warnings, you can read and save all the images in the dataset using alternative libraries, such as `skimage` or `Pillow`.

Similarly, the warnings from *torch.sigmoid* are also not important and will not halt the program; they can be safely ignored as well. The actual issue that is causing the program to stop may not be logged, as some GPU-related problems might not be printed in certain situations. To diagnose the issue, it might be helpful to try running another experiment using the CPU.

> GPU isn't being used for 'training'

According to your log file, it's important to ensure that the versions of `PyTorch` and `CUDA` are compatible with `PyTorch Lightning`. You can check this by running the command `python -c "import torch; print(torch.cuda.is_available())"` in your terminal. If the command returns `False`, you should consider installing the specific versions mentioned in the README file.

**Comment by mariam-162** [comment]
Thanks for the insight. torch.cuda.is_available() returns true. 
In the config.yaml file, I changed gpu: 0 and added gpus: 1. This is because I have a single GPU. So, training log has been updated here:
GPU available: True (cuda), used: True
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
HPU available: False, using: 0 HPUs
Trainer initailized.

Which is great. But after the same warnings in the training log attached above, the code exits.
Any leads to actually start training would be really appreciated.

Thanks for your time and effort once again.

**Comment by yiyulics** [comment]
I am not really sure about the actual reason of exiting without corresponding log. According to my experience, any strange exits could have something to do with OOM Errors. But I am not 100% sure. You may further check this out by adding extra log statements to monitor VRAM usage. Also, as I mentioned above, training on GPU sometimes hides real error information. You may try to run on CPU and it may throw real errors. After fixing the error, you can run on GPU safely.

**Comment by mariam-162** [comment]
Ok, I will dig deeper into the CPU version and share when I get somewhere. Thanks.

**Comment by yiyulics** [comment]
Closing due to inactivity. Feel free to reopen if you still have questions.
